src/chains/ethereum/ethereum/tests/api/eth/getBlockByNumber.test.ts

```diff
@@ -97,6 +99,299 @@ describe("api", () => {
           });
         });
       });
+
+      describe(`passing "pending" tag`, () => {
+        let from, to: string;
+        const emptyTransactionsPerBlock = 2;
+        const blockGasLimit = "0xA410"; // enough gas for 2 empty txs
+
+        /**
+         * Fetches a block with number `pendingBlock.number` and asserts that
+         * the resultant block is deep strict equal to the `pendingBlock`.
+         *
+         * NOTE: The `hash` and `timestamp` properties are removed from the
+         * block for comparison, since the `timestamp` of the block is largely
+         * dependent on CI speeds and the `hash` will be impacted by the
+         * `timestamp`.
+         * @param pendingBlock The pending block to compare.
+         */
+        const assertPendingEqualsMined = async (
```

Comment:
It's tricky without being able to interrogate the trie directly, but it'd be nice to be able to validate that mining the `pending` block doesn't affect the state of the trie.

Something like:
```javascript
const rootBefore = provider.#blockChain.trie.root.toString("hex");
//... create a pending block
const rootAfter = provider.#blockChain.trie.root.toString("hex");
assert.deepStrictEqual(rootBefore, rootAfter);
```

Reply:
I've added a test to do this, and it lead to me finding that we _were_ altering the state of things! Good suggestion. I've fixed this by adding `await vm.stateManager.checkpoint()`